### PR TITLE
EVAKA-4240 reservation calendar ux improvements

### PR DESCRIPTION
**Pull request by jaahon**
#### Summary

Fixes for some UX problems found in citizen calendar view. Includes partial streamlining of modals (FormModal and InfoModal) by making them fixed size and adding an updated close button to both of them. Also fixes some UX bugs in DatePicker. The rest of the chances should be reservation calendar specific.

Reviewing is probably the easiest when going through the commits one at a time.

### CODE REVIEW

**Comment by Joosakur** [comment]
started reviewing

File: frontend/src/citizen-frontend/calendar/CalendarGridView.tsx [hunk 1]
```diff
@@ -245,25 +245,15 @@ const DayCell = styled.div<{ today: boolean }>`
...
-    &:nth-child(6n+2) {
-      border-left: 1px solid ${colors.greyscale.lighter};
-    }
-      `}
-
-  /* top border for every day cell of first row */
-  &:nth-child(-n + 7) {
-    border-top: 1px solid ${colors.greyscale.lighter};
-  }
+      ? css`
```
Comment (Joosakur) [inline frontend/src/citizen-frontend/calendar/CalendarGridView.tsx:259]: 🤯

File: frontend/src/citizen-frontend/calendar/DayView.tsx [hunk 2]
```diff
@@ -70,122 +79,128 @@ export default React.memo(function DayView({
...
+                    <NoReservation>Ei varausta</NoReservation>
+                  )}
+                  <Label>Toteuma</Label>
+                  <span>–</span>
+                </Grid>
+              </Fragment>
+            ))}
+        {confirmationModal ? (
+          <InfoModal
+            title="Haluatko tallentaa muutokset?"
```
Comment (Joosakur) [inline frontend/src/citizen-frontend/calendar/DayView.tsx:184]: Should we already have some i18n for these?

Comment (jaahon) [reply frontend/src/citizen-frontend/calendar/DayView.tsx:184]: Probably, yeah.

**Review by Joosakur — APPROVED** [review]

**Comment by Joosakur** [comment]
minor things:

- select absence start date for next month, open end date picker, the initial month is current month (could use start date)
- creating reservations/absence reloads data and causes somewhat annoying scroll effect to happen again, but probably too difficult to fix to be wothwhile

**Comment by jaahon** [comment]
> * select absence start date for next month, open end date picker, the initial month is current month (could use start date)
Good catch, the initial month isn't updated because the date picker doesn't get rerendered when it's hidden.

> * creating reservations/absence reloads data and causes somewhat annoying scroll effect to happen again, but probably too difficult to fix to be wothwhile
I guess this is a UX question since we don't update data in place anywhere. I'll discuss this with the designers.

**Pull request merged**